source/api/content-state/0.9/index.md

```diff
@@ -0,0 +1,713 @@
+
+
+##### 2.3.2 Example of base64url encoding
+
+```json
+{
+  "id": "https://example.org/object1/canvas7#xywh=1000,2000,1000,2000",
+  "type": "Canvas",
+  "partOf": [{
+    "id": "https://example.org/object1/manifest",
+    "type": "Manifest"
+  }]
+}
+```
+
+The above annotation JSON can be condensed to remove unecessary whitespace:
+
+
+```
+{"id":"https://example.org/object1/canvas7#xywh=1000,2000,1000,2000","type":"Canvas","partOf":[{"id":"https://example.org/object1/manifest","type":"Manifest"}]}
+```
+
+The condensed form is then encoded (this example in Python):
+
+```python
+>>> import base64
+>>> base64.urlsafe_b64encode(b'{"id":"https://example.org/object1/canvas7#xywh=1000,2000,1000,2000","type":"Canvas","partOf":[{"id":"https://example.org/object1/manifest","type":"Manifest"}]}')
+
+b'eyJpZCI6ICJodHRwczovL2V4YW1wbGUub3JnL29iamVjdDEvY2FudmFzNyN4eXdoPTEwMDAsMjAwMCwxMDAwLDIwMDAiLCJ0eXBlIjoiQ2FudmFzIiwicGFydE9mIjpbeyJpZCI6Imh0dHBzOi8vZXhhbXBsZS5vcmcvb2JqZWN0MS9tYW5pZmVzdCIsInR5cGUiOiJNYW5pZmVzdCJ9XX0='
+```
+
+The string "eyJp...XX0=" is the now URI-safe, encoded form of the content state, suitable for passing to and from web applications.
+
+Not all use cases for providing a content state to a client require base64url encoding, because not all of them are susceptible to HTTP transmission issues. For example, a user interface could allow the user to paste or even type content state JSON-LD directly. Examples are given in section 3. In these scenarios, the client should accept unencoded JSON, too.
+
+
+#### 2.3.4 URI-Encoding
+
+If the content state is a simple URI, it _MUST NOT_ be base64url encoded. It _MAY_ be uri-encoded (percent encoding, as defined by [org-rfc-3986][Generic URI Syntax, RFC 3986]), and a client _MUST_ accept it in that form. 
+
+
+#### 2.3.4 Protocol
+
+Values are passed into web applications via request parameters, or attributes on HTML elements.
+
+If a client is capable of reading the content state from the value of an HTTP GET or POST request parameter, it _MUST_ look for the content state in a request parameter called `iiif-content`.
+
+If a client is capable of reading the content state from the value of an attribute on an HTML element that instantiates the client, it _SHOULD_ look for the content state in an attribute called `data-iiif-content`.
+
+A content state Annotation may be published inline as part of an HTML document, as in the preceding `data-iiif-content` example, or as a query string parameter forming part of the `href` property of an HTML anchor tag. 
+
+If the content state is a simple URI, the client _MUST_ load the resource at that URI and process it. The resource at that URI _MUST_ be the full Annotation as in 2.2.2. above, or a IIIF Resource as in 2.2.4. That is, the de-referenced response _MUST_ be JSON-LD, and _SHOULD_ have a value of `type` taken from `Annotation`, `Collection`, `Manifest`, `Canvas` and `Range`. The response _MUST_ use UTF-8 encoding.
```

Comment:
General comment:  Should references to examples by #links within the document?

Reply:
Yes probably - Presentation 3 does this. https://github.com/IIIF/api/blob/master/source/_includes/links.md has a load of links but they would have to go in now as 0.9 links and then be duplicated as 1.0 links. Maybe one to discuss.